libs/designer/src/lib/core/utils/outputs.ts

```diff
@@ -248,6 +259,89 @@ const convertSchemaAliasesForSplitOn = (schema: OpenAPIV2.SchemaObject): void =>
   }
 };
 
+export function convertDynamicExtensionsForSchema(schema: OpenAPIV2.SchemaObject): void {
+  if (!schema) {
+    return;
+  }
+
+  convertDynamicListExtension(schema);
+  convertDynamicTreeExtension(schema);
+  convertDynamicPropertiesExtension(schema);
+
+  convertDynamicExtensionsForSchemaChildren(schema);
+}
+
+function convertDynamicExtensionsForSchemaChildren(schema: OpenAPIV2.SchemaObject): void {
+  if (schema.properties) {
+    for (const childInputId of Object.keys(schema.properties)) {
+      const child = schema.properties[childInputId];
+
+      convertDynamicExtensionsForSchema(child);
+    }
+  }
+
+  if (schema.items) {
+    convertDynamicExtensionsForSchema(schema.items);
+  }
+}
+
+function convertDynamicListExtension(schema: OpenAPIV2.SchemaObject): void {
+  const dynamicListExtension = schema[ExtensionProperties.DynamicList];
+
+  if (!dynamicListExtension) {
+    return;
+  }
+
+  const originalDynamicState = dynamicListExtension.dynamicState;
+
+  if (originalDynamicState) {
+    schema[ExtensionProperties.DynamicList] = {
+      ...dynamicListExtension,
+      dynamicState: { ...originalDynamicState, schemaAlias: schema[ExtensionProperties.Alias] },
+    };
+  }
+}
+
+function convertDynamicTreeExtension(schema: OpenAPIV2.SchemaObject): void {
+  const dynamicTreeExtension = schema[ExtensionProperties.DynamicTree];
+
+  if (!dynamicTreeExtension) {
+    return;
```

Comment:
Shouldnt this be done at the service layer? We do not want to update the schema in designer, I know this is present in manifestService today because backend doesnt do the conversion, but this was kept in service because eventually backend needs to do this conversion and send it to client.

So please do not add this code here.

Reply:
Since the code for spliton lives in the designer, should we make the function getUpdatedManifestForSplitOn as a whole host-provided? A good chunk of the code (including incomplete aliasing code for spliton) lives here - is the ask to extract it all out to host?

Reply:
Pushed change to rely on host for updating the manifest or fall back to the current logic if not provided. This could likely be optimized but scoping the changes in the interest of unblocking PAuto on this issue ASAP.